src/networkmanager/exec.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sstream>
+#include <stdexcept>
+#include <thread>
+#include <vector>
+
+#include <Poco/Exception.h>
+#include <Poco/Pipe.h>
+#include <Poco/PipeStream.h>
+#include <Poco/Process.h>
+
+#include "exec.hpp"
+
+namespace aos::sm::cni {
+
+static Poco::Process::Env PrepareEnv(const std::string& args)
+{
+    Poco::Process::Env env;
+    std::istringstream iss(args);
+    std::string        token;
+
+    while (std::getline(iss, token, ' ')) {
+        if (!token.empty()) {
+            if (auto pos = token.find('='); pos != std::string::npos) {
+                std::string key   = token.substr(0, pos);
+                std::string value = token.substr(pos + 1);
+                env[key]          = value;
+            }
+        }
+    }
+
+    return env;
+}
+
+static std::string PluginErr(
+    const std::string& stderrContent, const std::string& stdoutContent, const std::string& defaultMessage)
+{
+    if (stdoutContent.empty()) {
+        if (stderrContent.empty()) {
+            return "plugin failed with no error message: " + defaultMessage;
```

Comment:
just `return defaultMessage` ? Otherwise, it will look long and not beauty.

Reply:
Done